metrics/coverage_converter/docs/01_coverage_converter_design.md

```diff
@@ -0,0 +1,148 @@
+
+As we can see above, the `CoverageConverterCommand` uses the `CoverageConverter` to convert the specific coverage report to the `CoverageModel` class that represents the `CI integrations` coverage report.
+
+Let's consider the activity diagram of the coverage conversion process: 
+
+![Coverage conversion diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/codecov_converter_design/metrics/coverage_converter/docs/diagrams/coverage_conversion_activity.puml)
+
+To run the command line application, we should have a `CommandRunner` class. Let's name it `CoverageConverterRunner`. This class will extend `CommandRunner`. It will be the place where we'll add our specific `CoverageConverterCommand`s.
+
+Also, we should be able to pass the following arguments to the Coverage Converter tool: 
+
+- input - the file from where we'll read the specific coverage report;
+- output - the file to which we'll write the formatted coverage report;
+
+To be able to pass arguments to our application we should create a `CoverageArgumentsParser` that will take a `List<String>` as arguments and return the `CoverageConverterArguments` instance that will contain all given arguments.
+
+## Package structure
+
+Once we've finished with all main classes and interfaces, let's consider the package structure of the `Coverage Converter`: 
+
+> lib
+>   - arguments
+>     - model
+>     - parser
+>   - common 
+>     - command
+>     - runner
+>     - converter
+>   - specific_format
+>     - command
+>     - model
+>     - converter
+
+
+## Creating a new specific format converter
+
+Once we have a `CoverageConverterCommand` abstract class, the `CoverageConverter` interface, and the `CoverageConverterRunner`,  we can create the first specific coverage parser. To implement it, we should follow the next steps: 
+
+1. Create a specific implementation of the `CoverageConverter`.
+2. Create a specific command extending the `CoverageConverterCommand` that will use the specific `CoverageConverter` created in the previous step.
+3. Add the previously created `CoverageConverterCommand` to the `CoverageConverterRunner` using the `addCommand` method.
+
+Currently, the Coverage Converter tool should support the following coverage report formats: 
+
+- LCOV
+- Istanbul
+
+## Summary
+
+Finally, let's consider the class diagram that provides information about all classes and their relationships.
+
+![Coverage class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/codecov_converter_design/metrics/coverage_converter/docs/diagrams/coverage_class_diagram.puml)
+
+So, the application takes the arguments, parses them using the `CoverageArgumentsParser`, and passes these arguments to the specific converter command. The converter command, in its turn, gets the input file path, converts it to the CI integrations coverage format, and writes it to the output file.
```

Comment:
Should we somehow cover what will happen if File not found? Or if we can't write output file (no disk space, etc..)?